firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -363,13 +364,14 @@ private PersistedInstallationEntry getPrefsWithGeneratedIdMultiProcessSafe() {
   }
 
   private String readExistingIidOrCreateFid(PersistedInstallationEntry prefs) {
-    // Check if this firebase app is the default (first initialized) instance
-    if (!firebaseApp.isDefaultApp() || !prefs.shouldAttemptMigration()) {
+    // Check if this firebase app is the default (first initialized) instance or is a chime app
+    if (!firebaseApp.getName().equals(CHIME_FIREBASE_APP_NAME)
+        && (!firebaseApp.isDefaultApp() || !prefs.shouldAttemptMigration())) {
       return fidGenerator.createRandomFid();
     }
-    // For a default firebase installation, read the existing iid from shared prefs
+    // For a default/chime firebase installation, read the existing iid from shared prefs
     String fid = iidStore.readIid();
-    if (fid == null) {
+    if (TextUtils.isEmpty(fid)) {
```

Comment:
shouldn't this be a separate PR?

Reply:
Small change and seemed relevant as the change the same method. Will do it next time.